main.js

```diff
@@ -1,6 +1,62 @@
-import './style.css'
+const APIKEY="api_key=f4fa6a2d7a3bb9c2777369817722c765" ;
+const BASE_URL="https://api.themoviedb.org/3";
+const API_URL=`${BASE_URL}/discover/movie?sort_by=popularity.desc&${APIKEY}`;
+const IMG="https://image.tmdb.org/t/p/w1280";
+const SEARCH =`${BASE_URL}/search/movie?&${APIKEY}`;
 
-document.querySelector('#app').innerHTML = `
-  <h1>Hello Vite!</h1>
-  <a href="https://vitejs.dev/guide/features.html" target="_blank">Documentation</a>
-`
+const noresult=document.querySelector('.noresult');
+const main =document.querySelector('.main');
+const movielist= document.querySelector('.movielist');
+const form= document.querySelector(".form");
+const search= document.querySelector(".search");
+
+async function getMovie(url){
+
+    const res=await fetch(url);
+    const data=await res.json();
+   
+    if(data.results.length>0){
+        showMovie(data.results.slice(0,15));
+        noresult.style.display="none";
+        movielist.style.display="block";
+    }
+    else{
+       
+        
+        noresult.style.display="block";
+        movielist.style.display="none";
+    }
+    //fetch(url).then(response => response.json).then(data=> console.log(data))
+}
+
+getMovie(API_URL);
+
+function showMovie(data){
+    console.log(data)
+    main.innerHTML=" ";
+    data.forEach(
+        function(result){
+            const imagePath =  IMG + result.poster_path;
+            
+        const {title}= result; 
+        const box=document.createElement('div');
+    box.classList.add("card");
+    box.innerHTML=`<img src="${imagePath}"  />
```

Comment:
An interesting conversation on why we shouldn't use innerHTML to create dom nodes through javascript - [Stackoverflow response](https://stackoverflow.com/questions/11515383/why-is-element-innerhtml-bad-code) Please refactor the code to not use innerHTML